lib/ide/src/controller.rs

```diff
@@ -0,0 +1,426 @@
+//! This module contains all the controllers. They cover everything that is
+//! between clients of remote services (like language server and file manager)
+//! and views.
+//!
+//! The controllers create a tree-like structure, with project controller being
+//! a root, then module controllers below, then graph/text controller and so on.
+//!
+//! As a general rule, while the "upper" (i.e. closer to root) nodes may keep
+//! handles to the "lower" nodes (e.g. to allow their reuse), they should never
+//! manage their lifetime.
+//!
+//! Primarily views are considered owners of their respective controllers.
+//! Additionally, controllers are allowed to keep strong handle "upwards".
+//!
+//! Controllers store their handles using `utils::cell` handle types to ensure
+//! that mutable state is safely accessed.
```

Comment:
I see that there was an effort to make these docs clear, and that is super cool. But TBH I dont understand it. There is a lot of questions:

- `This module contains all the controllers.` - what is a "controller"?
- "while the "upper" (i.e. closer to root) nodes" - what is node?
- "Primarily views are considered owners of their respective controllers." - I dont understand it at all. Whati s a primarily view?
- "Controllers store their handles" what are controller handles?

I see this is a complex topic. I feel that this is a great doc for someone who was implementing that but not for someone reading it without prior knowledge. For such a complex topic this doc sohuld be 2-3x longer and explain all the terms in a single place. I would be really thankful for that, cause I really want to understand in depth ewaht it is all about.

Reply:
This file has been created as a stubs for controllers, because they will be used by Views.

I'm refactoring this file in my task, so we may both remove this one, or add a TODO comment with information, that this is a stub and will be refactored

Reply:
Now I think that it will be better to just remove this file. I have already better copy on my branch, so there is no need to keep this stubs on master